pkg/auth/dependency/webapp/csrf_middleware.go

```diff
@@ -12,12 +13,30 @@ type CSRFMiddleware struct {
 }
 
 func (m *CSRFMiddleware) Handle(next http.Handler) http.Handler {
-	gorillaCSRF := csrf.Protect(
-		[]byte(m.Key),
-		csrf.Path("/"),
-		csrf.Secure(!m.UseInsecureCookie),
-		csrf.SameSite(csrf.SameSiteNoneMode),
-		csrf.CookieName(csrfCookieName),
-	)
-	return gorillaCSRF(next)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		useragent := r.UserAgent()
+		options := []csrf.Option{
+			csrf.Path("/"),
+			csrf.Secure(!m.UseInsecureCookie),
+			csrf.CookieName(csrfCookieName),
+		}
+		if samesite.ShouldSendSameSiteNone(useragent) {
+			options = append(options, csrf.SameSite(csrf.SameSiteNoneMode))
+		} else {
+			// http.Cookie SameSiteDefaultMode option will write SameSite
+			// with empty value to the cookie header which doesn't work for
+			// some old browsers
+			// To avoid writing samesite to the header
+			// set empty value to Cookie SameSite
+			// https://golang.org/src/net/http/cookie.go#L220
```

Comment:
I just searched and found this issue: https://github.com/golang/go/issues/36990
We may want to include the issue link for completeness.